src/map.cpp

```diff
@@ -1549,25 +1549,14 @@ ServerMapSector *ServerMap::createSector(v2s16 p2d)
 #endif
 
 	/*
-		Do not create over-limit.
-		We are checking for any nodes of the mapblocks of the sector being beyond the limit.
-		A sector is a vertical column of mapblocks, so sectorpos is like a 2D blockpos.
-
-		At the negative limit we are checking for
-			block minimum nodepos < -mapgenlimit.
-		At the positive limit we are checking for
-			block maximum nodepos > mapgenlimit.
-
-		Block minimum nodepos = blockpos * mapblocksize.
-		Block maximum nodepos = (blockpos + 1) * mapblocksize - 1.
+		Do not create over max mapgen limit
 	*/
-	const u16 map_gen_limit = MYMIN(MAX_MAP_GENERATION_LIMIT,
-		g_settings->getU16("map_generation_limit"));
-	if (p2d.X * MAP_BLOCKSIZE < -map_gen_limit
-			|| (p2d.X + 1) * MAP_BLOCKSIZE - 1 > map_gen_limit
-			|| p2d.Y * MAP_BLOCKSIZE < -map_gen_limit
-			|| (p2d.Y + 1) * MAP_BLOCKSIZE - 1 > map_gen_limit)
-		throw InvalidPositionException("createSector(): pos. over limit");
+	const s16 max_limit_bp = MAX_MAP_GENERATION_LIMIT / MAP_BLOCKSIZE;
```

Comment:
The previous code with `(p2d.X + 1) * MAP_BLOCKSIZE - 1 > map_gen_limit` etc was my own code and was necessary to fix the bugs caused by the problematic implementation of map gen limit. Before the map gen limit feature this code in createSector() was simpler and very similar to the new code in this PR.
Compare to https://github.com/minetest/minetest/blob/46684beec185d13f89c4a91aaa5dd2148ebb0273/src/map.cpp#L2503 which was the code before map gen limit was added.

Same applies elsewhere in this PR in mapblock.h, simpler code can be used because we are not struggling to cope with a bad implementation.